pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -277,5 +280,48 @@ func (s *Service) SDKToLaunchTemplate(d *ec2.LaunchTemplateVersion) (*expinfrav1
 		i.NetworkInterfaces = append(i.NetworkInterfaces, *tmp)
 	}
 
+	for _, id := range v.SecurityGroupIds {
+		i.AdditionalSecurityGroups = append(i.AdditionalSecurityGroups, infrav1.AWSResourceReference{ID: id})
```

Comment:
not a huge fan of the name `AdditionalSecurityGroups` when it's gunna include the core required security groups as well, we could rename the field to SecurityGroups but leave the json as additionalSecurityGroups, but I hate that too.

Reply:
i see what you mean. i vote we keep the json and name the same `AdditionalSecurityGroups`

Reply:
yeah, basically it is honestly only additional groups on the way in, so that part feels okay. i think i'll add a comment clarifying the behavior on the way _out_.